WordPress/gutenberg
# Refactor saveEntityRecord from redux-rungen to async thunks

Builds on top of the thunks support added in #27276 and refactors just the parts of core-data required to make the `saveEntityRecord` work (this PR is a minimal viable subset of #28389).

**Test plan:**

So you ask how to test this PR? The impact surface is pretty big. For sure confirm all the tests are green – that is a good indicator of the overall health. You may want to play with the Gutenberg too:
* Create a new post and save it, confirm the preview works
* Add a `blog title` block, edit it, save the post and related entities
* Play the widgets editor, add new widgets, update the existing ones
* Anything else that comes to your mind
